test/runner/xunit_reporter_test.dart

```diff
@@ -0,0 +1,332 @@
+             </failure>
+           </testcase>
+           <testcase classname="test.dart" name="success 1" time="0"> </testcase>
+           <testcase classname="test.dart" name="failure 2">
+             <failure message="oh no">
+               test.dart 8:33  main.&lt;fn&gt;
+             </failure>
+           </testcase>
+           <testcase classname="test.dart" name="success 2" time="0"> </testcase>
+        </testsuite>""");
+  });
+
+  test("always prints the full test name", () {
+    _expectReport(
+        """
+        test(
+           'really gosh dang long test name. Even longer than that. No, yet '
+               'longer. A little more... okay, that should do it.',
+           () {});""",
+        """
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <testsuite name="All tests" tests="1" errors="0" failures="0" skipped="0">
+           <testcase classname="test.dart" name="really gosh dang long test name. Even longer than that. No, yet longer. A little more... okay, that should do it." time="0"> </testcase>
+        </testsuite>""");
+  });
+
+  test("gracefully handles multiple test failures in a row", () {
+    _expectReport(
+        """
+        // This completer ensures that the test isolate isn't killed until all
+        // errors have been thrown.
+        var completer = new Completer();
+        test('failures', () {
+          new Future.microtask(() => throw 'first error');
+          new Future.microtask(() => throw 'second error');
+          new Future.microtask(() => throw 'third error');
+          new Future.microtask(completer.complete);
+        });
+        test('wait', () => completer.future);""",
+        """
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <testsuite name="All tests" tests="2" errors="1" failures="0" skipped="0">
+           <testcase classname="test.dart" name="failures">
+             <error message="first error">
+               test.dart 10:38  main.&lt;fn&gt;.&lt;fn&gt;
+               ===== asynchronous gap ===========================
+               dart:async       Future.Future.microtask
+               test.dart 10:15  main.&lt;fn&gt;
+             </error>
+             <error message="second error">
+               test.dart 11:38  main.&lt;fn&gt;.&lt;fn&gt;
+               ===== asynchronous gap ===========================
+               dart:async       Future.Future.microtask
+               test.dart 11:15  main.&lt;fn&gt;
+             </error>
+             <error message="third error">
+               test.dart 12:38  main.&lt;fn&gt;.&lt;fn&gt;
+               ===== asynchronous gap ===========================
+               dart:async       Future.Future.microtask
+               test.dart 12:15  main.&lt;fn&gt;
+             </error>
+           </testcase>
+           <testcase classname="test.dart" name="wait" time="0"> </testcase>
+        </testsuite>""");
+  });
+
+  test('update testSuite heading with correct number of tests', () {
```

Comment:
I haven't looked at the other tests in this package, so I don't know if they're using single or double quotes, but you have both in this file, so you should be consistent.